library/pk_wrap.c

```diff
@@ -1223,8 +1006,223 @@ static int ecdsa_sign_wrap(void *ctx, mbedtls_md_type_t md_alg,
                                          sig, sig_size, sig_len,
                                          f_rng, p_rng);
 }
+#endif /* MBEDTLS_USE_PSA_CRYPTO */
+#endif /* MBEDTLS_PK_CAN_ECDSA_SIGN */
+
+#if defined(MBEDTLS_PK_CAN_ECDSA_SOME)
+#if defined(MBEDTLS_ECP_RESTARTABLE)
```

Comment:
I think the parts about restartable ECDSA should all be guarded with `ECDSA_C && MBEDTLS_ECP_RESTARTABLE`, because that's what's actually needed for now.

I don't think `CAN_ECDSA_SOME && MBEDTLS_ECP_RESTARTABLE` is equivalent - for example you could have support for ECDSA in PSA, but not `ECDSA_C`, but `ECP_RESTARTABLE`, and then we wouldn't have what it takes to implement restartable ECDSA here.

Note: also, currently some places use `CAN_ECDSA_SOME && ECP_RESTARTABLE` while others use `ECDSA_C && ECP_RESTARTABLE` so at least we want consistency.